extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/VarBinaryChunkWriter.java

```diff
@@ -19,19 +18,188 @@
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
-import java.io.DataInput;
 import java.io.IOException;
 import java.io.OutputStream;
 import java.util.ArrayList;
-import java.util.Iterator;
-import java.util.PrimitiveIterator;
 
-public class VarBinaryChunkInputStreamGenerator<T> extends BaseChunkInputStreamGenerator<ObjectChunk<T, Values>> {
+public class VarBinaryChunkWriter<T> extends BaseChunkWriter<ObjectChunk<T, Values>> {
     private static final String DEBUG_NAME = "ObjectChunkInputStream Serialization";
     private static final int BYTE_CHUNK_SIZE = ChunkPoolConstants.LARGEST_POOLED_CHUNK_CAPACITY;
 
+    public interface Appender<T> {
+        void append(OutputStream out, T item) throws IOException;
+    }
+
     private final Appender<T> appendItem;
 
+    public VarBinaryChunkWriter(
+            final boolean fieldNullable,
+            final Appender<T> appendItem) {
+        super(ObjectChunk::isNull, ObjectChunk::getEmptyChunk, 0, false, fieldNullable);
+        this.appendItem = appendItem;
+    }
+
+    @Override
+    public DrainableColumn getInputStream(
+            @NotNull final ChunkWriter.Context<ObjectChunk<T, Values>> context,
+            @Nullable final RowSet subset,
+            @NotNull final BarrageOptions options) throws IOException {
+        return new ObjectChunkInputStream((Context) context, subset, options);
+    }
+
+    @Override
+    public Context makeContext(
+            @NotNull final ObjectChunk<T, Values> chunk,
+            final long rowOffset) {
+        return new Context(chunk, rowOffset);
+    }
+
+    public final class Context extends ChunkWriter.Context<ObjectChunk<T, Values>> {
+        private final ByteStorage byteStorage;
+
+        public Context(
+                @NotNull final ObjectChunk<T, Values> chunk,
+                final long rowOffset) {
+            super(chunk, rowOffset);
+
+            byteStorage = new ByteStorage(chunk.size() == 0 ? 0 : (chunk.size() + 1));
+
+            if (chunk.size() > 0) {
+                byteStorage.offsets.set(0, 0);
+            }
+
+            for (int ii = 0; ii < chunk.size(); ++ii) {
+                if (!chunk.isNull(ii)) {
+                    try {
+                        appendItem.append(byteStorage, chunk.get(ii));
```

Comment:
It feels like we are eagerly creating a buffer of all the data without respect to subset, which is different than our other writers which would use subset to make the buffer.

Reply:
In this context we're trying to perform any `toString` on source data one time for all subscribing clients rather than doing it once per client. I suspect this is appropriate.. but will think more about it.